src/dist/nfs/nfs_client_impl.cpp

```diff
@@ -502,5 +520,37 @@ void nfs_client_impl::handle_completion(const user_request_ptr &req, error_code
     // notify aio_task
     req->nfs_task->enqueue(err, err == ERR_OK ? total_size : 0);
 }
+
+void nfs_client_impl::register_cli_commands()
+{
+    _ctrl_nfs_max_copy_rate = dsn::command_manager::instance().register_app_command(
+        {"nfs.max_copy_rate"},
+        "nfs.max_copy_rate [num | DEFAULT]",
+        "control the max rate(MB/s) to copy file from remote node",
+        [this](const std::vector<std::string> &args) {
+            std::string result("OK");
+            if (args.empty()) {
+                result = std::to_string(_opts.max_copy_rate);
+            } else {
+                if (args[0] == "DEFAULT") {
+                    uint32_t burst_size = 1.5 * _opts.max_copy_rate * 1e6;
+                    _copy_token_bucket.reset(
```

Comment:
Whether _copy_token_bucket is thread safe or not？

Reply:
@Shuo-Jia Use BasicTokenBucket::reset(double genRate, double burstSize, double nowInSeconds) would be better than reset _copy_token_bucket pointer.